src/collectors/aws/ec2/instances.ts

```diff
@@ -40,14 +40,43 @@ export class EC2InstancesCollector extends BaseCollector {
                         fetchPending = false;
                     }
                 }
+                for (let instance of instances[region]) {
+                    if (instance.hasOwnProperty('BlockDeviceMappings')) {
+                        for (let blockDeviceMap of instance.BlockDeviceMappings) {
+                            blockDeviceMap.Ebs.Snapshots = await this.getSnapshotsByVolume(ec2, blockDeviceMap.Ebs.VolumeId);
+                        }
+                    }
+                }
             } catch (error) {
                 AWSErrorHandler.handle(error);
                 continue;
             }
         }
         return { instances };
-          // change instances
+        // change instances
 
     }
 
+    private async getSnapshotsByVolume(ec2Obj: AWS.EC2, volumeId: string) {
+        let snapshots = [] as any;
+        let fetchPending = true;
+        let marker: string | undefined;
+        while (fetchPending) {
+            const params: AWS.EC2.DescribeSnapshotsRequest = {
+                Filters: [
+                    {
+                        Name: 'volume-id',
```

Comment:
Use start-time filter, otherwise, there might be many snapshots. refer https://github.com/tensult/cloud-reports/blob/master/src/collectors/aws/ebs/volume_snapshots.ts